models/user.go

```diff
@@ -520,16 +530,18 @@ func (u *User) UploadAvatar(data []byte) error {
 	}
 
 	u.UseCustomAvatar = true
-	u.Avatar = fmt.Sprintf("%x", md5.Sum(data))
+	u.Avatar = fmt.Sprintf("%v-%x", u.ID, md5.Sum(data))
 	if err = updateUser(sess, u); err != nil {
 		return fmt.Errorf("updateUser: %v", err)
 	}
 
-	if err := os.MkdirAll(setting.AvatarUploadPath, os.ModePerm); err != nil {
-		return fmt.Errorf("Failed to create dir %s: %v", setting.AvatarUploadPath, err)
+	fs := storage.FileStorage{
+		Ctx:      context.Background(),
+		Path:     setting.AvatarUploadPath,
+		FileName: u.Avatar,
```

Comment:
Same question as for repo avatar file name.

Reply:
Same answer as for repo.avatar.
But previously, if two users uploaded the same file, it would replace the previous one (also in previous system).
https://github.com/go-gitea/gitea/blob/d4cd4ed4422be222088dd8535a228b906f43fdc2/models/user.go#L523

I used same format as `repo.Avatar` to avoid this collision.
https://github.com/go-gitea/gitea/blob/98c0c8ea2e0d4896cc3cc00423f0d2952fcbad21/models/user.go#L533